src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -100,94 +100,37 @@ const checkIsSpam = (tx: Tx): boolean => {
   })
 }
 
-/**
- * Manage state of the txHistory slice
- *
- * If transaction already exists, update the value, otherwise add the new transaction
- */
-
-const updateOrInsertTx = (txHistory: TxHistory, tx: Tx, accountId: AccountId) => {
-  const { txs, hydrationMeta } = txHistory
-
-  if (checkIsSpam(tx)) return
-
-  const txIndex = serializeTxIndex(accountId, tx.txid, tx.pubkey, tx.data)
-
-  const isNew = !txs.byId[txIndex]
-
-  // update or insert tx
-  txs.byId[txIndex] = tx
-
-  // add id to ordered set for new tx
-  if (isNew) {
-    const orderedTxs = orderBy(txs.byId, 'blockTime', ['desc'])
-    const index = orderedTxs.findIndex(
-      tx => serializeTxIndex(accountId, tx.txid, tx.pubkey, tx.data) === txIndex,
-    )
-    txs.ids.splice(index, 0, txIndex)
-  }
-
-  // for a given tx, find all the related assetIds, and keep an index of
-  // txids related to each asset id
-  getRelatedAssetIds(tx).forEach(relatedAssetId =>
-    deepUpsertArray(txs.byAccountIdAssetId, accountId, relatedAssetId, txIndex),
-  )
-
-  // update the min tx blocktime
-  const hydrationMetadataForAccountId = hydrationMeta[accountId]
-  if (
-    hydrationMetadataForAccountId === undefined ||
-    tx.blockTime < (hydrationMetadataForAccountId.minTxBlockTime ?? Infinity)
-  ) {
-    hydrationMeta[accountId] = {
-      isHydrated: false,
-      isErrored: false,
-      minTxBlockTime: tx.blockTime,
-    }
-  }
-}
-
 const updateOrInsertTxs = (txHistory: TxHistory, incomingTxs: Tx[], accountId: AccountId) => {
   const { txs, hydrationMeta } = txHistory
 
-  const filteredIncomingTxsWithIndex = incomingTxs
-    .filter(tx => !checkIsSpam(tx))
-    .map(tx => {
-      const txIndex = serializeTxIndex(accountId, tx.txid, tx.pubkey, tx.data)
-      return { tx, txIndex }
-    })
-
+  let newTxAdded = false
   let minTxBlockTime = Infinity
+  for (const tx of incomingTxs) {
+    if (checkIsSpam(tx)) continue
+    if (tx.blockTime < minTxBlockTime) minTxBlockTime = tx.blockTime
 
-  for (const { txIndex, tx } of filteredIncomingTxsWithIndex) {
-    if (minTxBlockTime > tx.blockTime) {
-      minTxBlockTime = tx.blockTime
-    }
-
-    const isNew = !txs.byId[txIndex]
+    const txIndex = serializeTxIndex(accountId, tx.txid, tx.pubkey, tx.data)
 
-    if (!isNew) continue
+    // track if new transactions have been added to prompt (re)sort of ids
+    if (newTxAdded === false && !txs.byId[txIndex]) newTxAdded = true
```

Comment:
Here she is!

Reply:
This part is actually just to prevent unnecessary resort of ids if we haven't actually added any new transactions for minor perf improvement.

The fix was removal of `if (!isNew) continue` which was preventing upsert of existing txs.